oap-server/server-receiver-plugin/skywalking-trace-receiver-plugin/src/test/java/org/apache/skywalking/oap/server/receiver/trace/provider/handler/v8/rest/TraceSegmentReportServletHandlerTest.java

```diff
@@ -0,0 +1,127 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.oap.server.receiver.trace.provider.handler.v8.rest;
+
+import org.apache.skywalking.apm.network.language.agent.v3.SegmentObject;
+import org.apache.skywalking.oap.server.library.module.ModuleManager;
+import org.apache.skywalking.oap.server.telemetry.TelemetryModule;
+import org.apache.skywalking.oap.server.telemetry.api.MetricsCreator;
+import org.apache.skywalking.oap.server.telemetry.none.MetricsCreatorNoop;
+import org.apache.skywalking.oap.server.telemetry.none.NoneTelemetryProvider;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.internal.util.reflection.Whitebox;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import javax.servlet.http.HttpServletRequest;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.List;
+
+import static org.mockito.Mockito.when;
+
+@RunWith(PowerMockRunner.class)
+@PowerMockIgnore({"javax.management.*"})
+public class TraceSegmentReportServletHandlerTest {
+
+    @Mock
+    private HttpServletRequest request;
+    @Mock
+    private ModuleManager moduleManager;
+    @Mock
+    private NoneTelemetryProvider telemetryProvider;
+
+    @Before
+    public void init() throws IOException {
+        TelemetryModule telemetryModule = Mockito.spy(TelemetryModule.class);
+        Whitebox.setInternalState(telemetryModule, "loadedProvider", telemetryProvider);
+        Mockito.when(moduleManager.find(TelemetryModule.NAME)).thenReturn(telemetryModule);
+
+        Mockito.when(telemetryProvider.getService(MetricsCreator.class))
+            .thenReturn(new MetricsCreatorNoop());
+    }
+
+    @Test
+    public void testSingle() throws IOException {
+        String singleJson = "{" +
+            "   \"traceId\":\"c480c738-b628-490d-ace7-69f7030d77cb\"," +
+            "   \"spans\":[" +
+            "       {\"operationName\":\"\\/ingress\"}" +
+            "   ]" +
+            "}";
+
+        final TraceSegmentReportSingleServletHandler singleServletHandler =
+            new TraceSegmentReportSingleServletHandler(moduleManager, null, null);
+
+        when(request.getReader()).thenReturn(new BufferedReader(new StringReader(singleJson)));
+        final List<SegmentObject> segmentObjects = singleServletHandler.parseSegments(request);
+        Assert.assertEquals(segmentObjects.size(), 1);
+    }
+
+    @Test
+    public void testMultipleLine() throws IOException {
+        String multipleLineJson = "{" +
```

Comment:
I don't think this is a good example. This is not an official format in JSON.

Reply:
And isn't better than the official JSON list way.

Reply:
It adapts to the current LUA agent. Should I need to fix the LUA first? let the LUA agent using the JSON object or JSON list.